### PR TITLE
REFAPP-288 Custom header validation run

### PR DESCRIPTION
**Pull request by ezodude**
__UPDATE__
* Intermittent test failure cause found and fixed.

This PR adds the custom`x-validation-run-id` to response logging. This is required for downstream validation and reporting.

In fact `setupResponseLogging(...)` demands the header is present:
```
assert.ok(extras.validationRunId, 'validationRunId missing from setupResponseLogging call');
```

But this is open for debate.

### CODE REVIEW

File: app/ob-util.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ const setHeaders = (requestObj, headers) => {
   setOptionalHeader('x-fapi-customer-last-logged-time', headers.customerLastLogged, requestObj);
   setOptionalHeader('x-fapi-customer-ip-address', headers.customerIp, requestObj);
   setOptionalHeader('x-jws-signature', headers.jwsSignature, requestObj);
+  setOptionalHeader('x-validation-run-id', headers.jwsSignature, requestObj);
```
Comment (robmckinnon) [inline app/ob-util.js]: Should be `headers.validationRunId`.

File: app/response-logger.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ const logger = bunyan.createLogger({
 
 const setupResponseLogging = (requestObj, extras) => {
   if (process.env.NODE_ENV !== 'test' && process.env.LOG_ASPSP_RESPONSES === 'true') {
+    assert.ok(extras.validationRunId, 'validationRunId missing from setupResponseLogging call');
```
Comment (robmckinnon) [inline app/response-logger.js]: Think the place to add this check is in [this file where we do the kafka logging](https://github.com/OpenBankingUK/tpp-reference-server/blob/es-refapp-288-custom-header-validation-run-id/app/validator/validate-request-response.js#L62).

It a bit confusing, but the response-logger.js is not where the kafka logging is happening. It has the original logging to file implementation only.

As it's optional, this test for validationRunId in response-logger.js probably not necessary. Suggest we remove it to enable people to use file logging when not doing validation.

**Comment by ezodude** [comment]
@robmckinnon thanks for the heads up. Corrections made.

File: test/session/request-headers.test.js [hunk 3]
```diff
@@ -24,26 +25,34 @@ const { extractHeaders } = proxyquire(
...
 };
 
 describe('extractHeaders from request headers', () => {
-  it('returns authorisationServerId and headers object', async () => {
+  it('returns headers object', async () => {
     const interactionId = generatedInteractionId;
     const headers = await extractHeaders(requestHeaders);
-    // assert.equal(value.authorisationServerId, authorisationServerId);
-    // assert.equal(value.authorisationServerId, authorisationServerId);
     assert.deepEqual(headers, {
```
Comment (mbana) [inline test/session/request-headers.test.js]: perhaps make this into a common var that is used in `'returns headers with same interactionId'` as well?
something like `HEADERS_EXPECTED`?

Comment (ezodude) [reply test/session/request-headers.test.js]: Taken into account below.

**Pull request merged**